### PR TITLE
Docblock Improvements + fix for ActivityApi::addNote Result

### PR DESCRIPTION
**Pull request by wodka**
- add exceptions to each docblock
- add return types
- add boolean for successful delete in api
- fix ActivityApi::addNote return type

### CODE REVIEW

**Comment by mavimo** [comment]
@wodka can you rebase it, please?

**Comment by wodka** [comment]
I will do that - might just take a few weeks till I have time

**Comment by mavimo** [comment]
@wodka ok, thanks, let me know when it's done.

**Comment by mavimo** [comment]
@wodka do you think to be able to rebase this week? let me know in order to include it on release 0.7 or in the next release.

**Comment by mavimo** [comment]
@wodka with PHP7 we run also `phpstan` for static analysis, this tool actually report that the annotation reported do not match with the current code. The reported errors are:
```

 ------ ------------------------------------------------------------------------- 
  Line   src/LooplineSystems/CloseIoApiWrapper/CloseIoApiWrapper.php              
 ------ ------------------------------------------------------------------------- 
  72     Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getLeadApi()        
         should return LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi  
         but returns null.                                                        
  84     Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getCustomFieldApi(  
         ) should return                                                          
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi but returns    
         null.                                                                    
  96     Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getOpportunityApi(  
         ) should return                                                          
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi but returns    
         null.                                                                    
  108    Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getLeadStatusesApi  
         () should return                                                         
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi but returns    
         null.                                                                    
  120    Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getOpportunityStat  
         usesApi() should return                                                  
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi but returns    
         null.                                                                    
  132    Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getContactApi()     
         should return LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi  
         but returns null.                                                        
  144    Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getActivitiesApi()  
         should return LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi  
         but returns null.                                                        
  156    Method                                                                   
         LooplineSystems\CloseIoApiWrapper\CloseIoApiWrapper::getTaskApi()        
         should return LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi  
         but returns null.                                                        
 ------ ------------------------------------------------------------------------- 
 ------ ----------------------------------------------------------------------- 
  Line   tests/LooplineSystems/CloseIoApiWrapper/Library/Curl/CurlTest.php      
 ------ ----------------------------------------------------------------------- 
  47     Call to an undefined method                                            
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi::getLead().  
 ------ ----------------------------------------------------------------------- 
 ------ ------------------------------------------------------------------------- 
  Line   tests/LooplineSystems/CloseIoApiWrapper/Api/CustomFieldApiTest.php       
 ------ ------------------------------------------------------------------------- 
  94     Method                                                                   
         Tests\LooplineSystems\CloseIoApiWrapper\Api\CustomFieldApiTest::getCust  
         omFieldApi() should return                                               
         LooplineSystems\CloseIoApiWrapper\Api\CustomFieldApi but returns         
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi.               
 ------ ------------------------------------------------------------------------- 
 ------ ------------------------------------------------------------------------- 
  Line   tests/LooplineSystems/CloseIoApiWrapper/Api/ActivityApiTest.php          
 ------ ------------------------------------------------------------------------- 
  77     Method                                                                   
         Tests\LooplineSystems\CloseIoApiWrapper\Api\ActivityApiTest::getActivit  
         yApi() should return LooplineSystems\CloseIoApiWrapper\Api\ActivityApi   
         but returns LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi.   
 ------ ------------------------------------------------------------------------- 
 ------ ------------------------------------------------------------------------- 
  Line   tests/LooplineSystems/CloseIoApiWrapper/Api/LeadsApiTest.php             
 ------ ------------------------------------------------------------------------- 
  171    Method                                                                   
         Tests\LooplineSystems\CloseIoApiWrapper\Api\LeadsApiTest::getLeadsApi()  
         should return LooplineSystems\CloseIoApiWrapper\Api\LeadApi but returns  
         LooplineSystems\CloseIoApiWrapper\Library\Api\AbstractApi.               
 ------ ------------------------------------------------------------------------- 
 ------ ------------------------------------------------------------------------ 
  Line   tests/LooplineSystems/CloseIoApi/Api/ConfigTest.php                     
 ------ ------------------------------------------------------------------------ 
         Class LooplineSystems\CloseIoApiWrapper\Tests\ConfigTest was not found  
         while trying to analyse it - autoloading is probably not configured     
         properly.                                                               
 ------ ------------------------------------------------------------------------ 
                                                                                
 [ERROR] Found 13 errors                                                        
```

File: tests/LooplineSystems/CloseIoApiWrapper/Api/ConfigTest.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * @copyright Copyright (c) 2014 LLS Internet GmbH - Loopline Systems (http://www.loopline-systems.com)
+ * @license   https://github.com/loopline-systems/closeio-api-wrapper/blob/master/LICENSE (MIT Licence)
+ */
+
+namespace Tests\LooplineSystems\CloseIoApiWrapper\Api;
+
+use LooplineSystems\CloseIoApiWrapper\CloseIoConfig;
+use LooplineSystems\CloseIoApiWrapper\Library\Exception\InvalidParamException;
+
+class ConfigTest extends \PHPUnit_Framework_TestCase
```
Comment (mavimo) [inline tests/LooplineSystems/CloseIoApiWrapper/Api/ConfigTest.php]: Should integrate this tests into https://github.com/loopline-systems/closeio-api-wrapper/blob/develop/tests/LooplineSystems/CloseIoApiWrapper/CloseIoConfigTest.php instead in a new file? Thanks!

**Pull request merged**